src/Hearthrock/Engine/RockEngineAction.cs

```diff
@@ -138,13 +138,16 @@ public void ApplyAll()
         {
             foreach (int cardId in this.actions)
             {
-                var card = this.pegasus.GetObject(cardId);
-                if (card.CardId == "GAME_005")
+                var card = this.pegasus?.GetObject(cardId);
```

Comment:
this.pegasus can be null? It doesn't make sense.
I prefer to undo the changes in here, if "this.pegasus" could be null, I should fix it.

BTW I prefer to fix null instead of tolerating null. Because I prefer to have fewer states. An uncontrollable number of states usually lead to bugs.

Reply:
it's happens sometime, and mulligan do not works as expected, I do not why it happens, it's really strange. yes sure I will undo this, maybe we will find the real reason.